libs/wire-api/src/Wire/API/Team/Feature.hs

```diff
@@ -344,7 +354,41 @@ data FeatureTTL
   = FeatureTTLSeconds Word
   | FeatureTTLUnlimited
   deriving stock (Eq, Show, Generic)
-  deriving (Arbitrary) via (GenericUniform FeatureTTL)
+
+instance Arbitrary FeatureTTL where
+  arbitrary = nonZero <$> arbitrary
+    where
+      nonZero 0 = FeatureTTLUnlimited
+      nonZero n = FeatureTTLSeconds n
+
+instance ToSchema FeatureTTL where
+  schema = mkSchema ttlDoc toTTL fromTTL
+    where
+      ttlDoc :: NamedSwaggerDoc
+      ttlDoc = swaggerDoc @Word & S.schema . S.example ?~ "unlimited"
+
+      toTTL :: A.Value -> A.Parser FeatureTTL
+      toTTL v = parseUnlimited v <|> parseSeconds v
+
+      parseUnlimited :: A.Value -> A.Parser FeatureTTL
+      parseUnlimited =
+        A.withText "FeatureTTL" $
+          \t ->
+            if t == "unlimited" || t == "0"
+              then pure FeatureTTLUnlimited
+              else A.parseFail "Expected ''unlimited' or '0'."
+
+      parseSeconds :: A.Value -> A.Parser FeatureTTL
+      parseSeconds = A.withScientific "FeatureTTL" $
+        \s -> case toBoundedInteger s of
+          Just 0 -> A.parseFail "Expected a positive integer. Got 0 instead."
```

Comment:
```suggestion
          Just 0 -> error "impossible (this would have parsed in `parseUnlimited` above)."
```

Otherwise I think this is fine.  (I said I wanted to look at this un-isomorphismish-ness again, but I think we should leave it as is.)